### PR TITLE
Projections Update

### PR DESCRIPTION
**Pull request by prolic**

### CODE REVIEW

File: src/Projection/PDOEventStoreQuery.php [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+    /**
+     * @var int lock timeout in milliseconds
+     */
+    private $lockTimeoutMs;
+
+    /**
+     * @var int
+     */
+    private $eventCounter = 0;
+
```
Comment (sandrokeil) [inline src/Projection/PDOEventStoreQuery.php]: `$lockTimeoutMs` and `$eventCounter` is never used!?

File: src/Projection/PDOEventStoreProjectionFactory.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\PDO\Projection;
+
+use PDO;
```
Comment (sandrokeil) [inline src/Projection/PDOEventStoreProjectionFactory.php]: `use PDO` is never used. Please remove

**Comment by sandrokeil** [comment]
The parameter type description here `\Prooph\EventStore\PDO\PDOStreamIterator::$currentItem` is wrong. `array|false` and access with `$this->currentItem->created_at`?

**Comment by sandrokeil** [comment]
Using always camel case looks better to me e.g. `PDO` -> `Pdo` or `MySQL` -> `MySql`. Because `PDOEventStoreException` is *hard* to read. What do you think?

File: src/Projection/PDOEventStoreProjectionFactory.php (hunk 2 above)
Comment (prolic) [inline src/Projection/PDOEventStoreProjectionFactory.php]: it's used in the constructor

**Comment by prolic** [comment]
@sandrokeil updated

**Comment by prolic** [comment]
@codeliner @sandrokeil @oqq about naming:

I am not sure about the last commit, the base class in PHP is PDO, not Pdo. Also Doctrine uses this f.e. https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Platforms/MySQL57Platform.php

**Comment by oqq** [comment]
There are several comments to change this in php code and also in doctrine. Especially in doctrine code, cuz the PDO prefix not fits to PSR-1.

I am for Pdo instead of PDO since 1 year for that PSR reasons and have used PDO before. But it´s a matter of taste, so no hard opinion from me.

**Comment by bweston92** [comment]
Can it be explained why we are coupling a read model with the event store?

**Comment by prolic** [comment]
@bweston92 I don't get your question. The read model is a projection of the recorded events. So in order to create a readmodel, you have to get a stream of events and process it. Nothing else. The readmodel itself has no dependency on the event store. The projector needs to have knowledge about the event store and the read model, in order to create it.

**Comment by bweston92** [comment]
@prolic sorry I just seen read model and projection key words in the event store.

**Pull request merged**